### PR TITLE
Baking in package to mfsbsd image

### PR DESCRIPTION
**Pull request by bretton**
This adds packages to mfsbsd as part of the build process. 

The documented approach is out of date. 

Simply copy mfsbsd/tools/packages.sample to mfsbsd/tools/packages

or copy in own file to mfsbsd/tools/packages

### CODE REVIEW

**Comment by bretton** [comment]
local build and run test works
test on hetzner ipv6 only works with some issues, won't prefer AAAA records. `pkg -6` works

File: .gitignore [hunk 1]
```diff
@@ -14,5 +14,8 @@ authorized_keys_in
 mfsbsd/conf/rc.local
 mfsbsd/conf/rc.conf
 mfsbsd/conf/loader.conf
+mfsbsd/*.gz
+mfsbsd/*.iso
+mfsbsd/*.img
```
Comment (grembo) [inline .gitignore:19]: Hm, those are files in a submodule - does it really make sense to exclude them (didn't test if they show up - if this works, ignore my comment)?

File: build.sh [hunk 2]
```diff
@@ -9,6 +9,7 @@
 # 2022-08-05: adjustments based on github feedback
 # 2022-08-12: bsdinstall customisations
 # 2022-08-15: some general improvements
+# 2022-08-23: include necessary packages in the mfsbsd image
```
Comment (grembo) [inline build.sh:12]:
```suggestion
# 2022-08-23: include necessary packages in the mfsbsd image
#             add enable_ipv6.sh script
```

File: customfiles/enable_ipv6.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+service dhcpcd enable
+service dhcpcd start
+sysrc ip6addrctl_policy=ipv6_prefer
+service ip6addrctl start
```
Comment (grembo) [inline customfiles/enable_ipv6.sh]: This worked for me, even though it's a bit ugly.

```suggestion
#!/bin/sh

killall dhclient

cat >/usr/local/etc/dhcpcd.conf <<EOF
duid
persistent
vendorclassid
option interface_mtu
option rapid_commit
slaac private
ipv6only
ipv6rs
nodhcp6
waitip 6
EOF

service dhcpcd enable
service dhcpcd restart
sysrc ip6addrctl_policy=ipv6_prefer
service ip6addrctl start
route delete default
```

File: README.md [hunk 4]
```diff
@@ -31,6 +31,11 @@ the `-u` flag is added to `build.sh`.
 ### Configure your local customisations
 Optionally edit the files in `customfiles/*` to make flavour changes for `depenguin-me`. 
 
+### Configure custom packages
+You can edit the file `customfiles/depenguin_packages.txt` if you have additional packages to include. 
+
+Beware file size of resulting image is a potential concern, see MFSBSD documentation.
```
Comment (grembo) [inline README.md]:
```suggestion
Beware file size of resulting image is a potential concern, see mfsBSD documentation.
```

**Review by grembo — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (bretton) [reply .gitignore:19]: those are built files, this change was from few days back, forgot it was included. should be fine to leave it

File: build.sh [hunk 5]
```diff
@@ -180,6 +183,12 @@ for bsdinstall_file in "${custom_bsdinstall_files[@]}"; do
 	fi
 done
 
+# add a list of packages to bake into the image
+if [ -f "$MYCUSTOMDIR/depenguin_packages.txt" ]; then
+    cp -f "$MYCUSTOMDIR/depenguin_packages.txt" "$BASEDIR/mfsbsd/tools/packages"
```
Comment (grembo) [inline build.sh]:
```suggestion
	 cp -f "$MYCUSTOMDIR/depenguin_packages.txt" "$BASEDIR/mfsbsd/tools/packages"
```

File: build.sh [hunk 6]
```diff
@@ -180,6 +183,12 @@ for bsdinstall_file in "${custom_bsdinstall_files[@]}"; do
...
+else
+    exit_error "missing packages file"
```
Comment (grembo) [inline build.sh]:
```suggestion
	 exit_error "missing packages file"
```

**Review by grembo — CHANGES_REQUESTED** [review]

**Review by grembo — APPROVED** [review]

**Pull request merged**